src/components/input/tests/Input.spec.tsx

```diff
@@ -31,6 +31,47 @@ describe('Input', () => {
       input.detach();
     });
 
+    it('should set inner input classes when specified', () => {
+      const innerInputClass = 'valid';
+      const classes = [innerInputClass];
+      const innerInput = input.find('input').first();
+      expect(innerInput.hasClass(innerInputClass)).toBe(false);
+
+      input.setProps({ innerInputClasses: classes });
+      input.mount();
+      expect(innerInput.hasClass(innerInputClass)).toBe(true);
+    });
+
+    it('should set inner input type when specified', () => {
+      const type = 'password';
+      const innerInput = input.find('input').first();
+      expect(innerInput.prop('type')).toBe('text');
+
+      input.setProps({ type });
+      input.mount();
+      expect(innerInput.prop('type')).toBe(type);
+    });
+
+    it('should set inner label valid message when specified', () => {
+      const message = 'salut';
+      const innerLabel = input.find('label').first();
+      expect(innerLabel.prop('data-valid-message')).toBe(undefined);
+
+      input.setProps({ validMessage: message });
+      input.mount();
+      expect(innerLabel.prop('data-valid-message')).toBe(message);
+    });
+
+    it('should set inner label invalid message when specified', () => {
+      const message = 'salut';
+      const innerLabel = input.find('label').first();
+      expect(innerLabel.prop('data-invalid-message')).toBe(undefined);
+
+      input.setProps({ invalidMessage: message });
+      input.mount();
```

Comment:
you can chain the calls and mount the input right after setting the props:

`input.setProps({ invalidMessage: message }).mount();`